dev/com.ibm.ws.security.oidc.client_fat.4/fat/src/com/ibm/ws/security/openidconnect/client/fat/IBM/OidcClientCallerClaims.java

```diff
@@ -0,0 +1,647 @@
+
+    }
+
+
+    /**
+     * testing tokenOrderTofetchCallerClaims="IDToken"
+     * caller claim (upn) does not exist in any token
+     * 
+     * @throws Exception
+     */   
+    @Test
+    public void IDTokens_user_claim_missing() throws Exception {
+
+        String appName = "idtokenonly";
+
+        WebConversation wc = new WebConversation();
+        TestSettings updatedTestSettings = testSettings.copyTestSettings();
+        updatedTestSettings.setScope("openid profile");
+        updatedTestSettings.setTestURL(testSettings.getTestURL().replace("SimpleServlet", "simple/" + appName));
+
+        List<validationData> expectations = vData.addSuccessStatusCodes(null, Constants.LOGIN_USER);
+       
+        expectations = vData.addResponseStatusExpectation(expectations, Constants.LOGIN_USER, com.ibm.ws.security.fat.common.Constants.UNAUTHORIZED_STATUS);
+
+        expectations = validationTools.addMessageExpectation(testRPServer, expectations, Constants.LOGIN_USER, Constants.MESSAGES_LOG, Constants.STRING_CONTAINS, "Client messages.log should contain a message indicating that client failed to authenticate the JSON Web Token", MessageConstants.CWWKS1738E_JWT_MISSING_CLAIM);
+ 
+        JWTTokenBuilder builder = createBuilderWithDefaultClaims();
+        builder.unsetClaim("upn"); //unset userIdentifier = upn
+        String jwtTokenWithNoUserClaim = builder.build();
+
+        // the built token will be passed to the test app via the overrideToken parm - it will be saved to be later returned during the auth process.
+        List<endpointSettings> parms = eSettings.addEndpointSettingsIfNotNull(null, "overrideToken", jwtTokenWithNoUserClaim);
+        parms = eSettings.addEndpointSettingsIfNotNull(parms, "overrideIDToken", jwtTokenWithNoUserClaim);
+        genericInvokeEndpointWithHttpUrlConn(_testName, null, updatedTestSettings.getTokenEndpt(), Constants.PUTMETHOD, "misc", parms, null, expectations);
+  
+        List<endpointSettings> userinfParms = eSettings.addEndpointSettingsIfNotNull(null, "userinfoToken", jwtTokenWithNoUserClaim);
+        //(That url that the RP will call is:  http://localhost:${bvt.prop.security_1_HTTP_default}/UserinfoEndpointServlet/getJWT)
+        genericInvokeEndpointWithHttpUrlConn(_testName, null, updatedTestSettings.getUserinfoEndpt(), Constants.PUTMETHOD, "misc", userinfParms, null, expectations);
+        // we created and saved tokens for our test tooling token endpoint and userinfo endpoint to return to the RP - let's invoke
+        // the protected resource.  The RP will get the auth token, but, instead of getting at, idt and userinfo from the OP, it will use a
+        // token ep and userinfo ep pointing to the test tooling app that will return the tokens previously obtained using a builder
+        genericRP(_testName, wc, updatedTestSettings, Constants.GOOD_OIDC_LOGIN_ACTIONS_SKIP_CONSENT, expectations);
+
+    }
+
+
```

Comment:
1. I would suggest adding some tests that contain the claim that is being tested with different/alt values.  If the search order is TOKEN_ORDER_ACCESSTOKEN_IDTOKEN_USERINFO, have access_token contain the claim as expected and then have id_token and userinfo contain some other value (cases with different values and others with a superset of the value in the access_token)
2. Have configs with different multiple token orders - such as TOKEN_ORDER_ACCESSTOKEN_IDTOKEN_USERINFO and TOKEN_ORDER_USERINFO_IDTOKEN_ACCESSTOKEN, ...